File: accommodation_service/src/main/java/huy/project/accommodation_service/core/port/ILanguagePort.java
package huy.project.accommodation_service.core.port;

import huy.project.accommodation_service.core.domain.dto.request.GetLanguageParams;
import huy.project.accommodation_service.core.domain.dto.response.PageInfo;
import huy.project.accommodation_service.core.domain.entity.LanguageEntity;
import org.springframework.data.util.Pair;

import java.util.List;

public interface ILanguagePort {
    LanguageEntity save(LanguageEntity language);
    LanguageEntity getLanguageByName(String name);
    LanguageEntity getLanguageByCode(String code);
    Pair<PageInfo, List<LanguageEntity>> getLanguages(GetLanguageParams params);
    void deleteLanguageById(Long id);
    LanguageEntity getLanguageById(Long id);
    List<LanguageEntity> getLanguagesByIds(List<Long> ids);
    long countAll();
}

File: location_service/LocationService.Core/Port/ILanguagePort.cs
namespace LocationService.Core.Port
{
    using System.Threading.Tasks;
    using LocationService.Core.Domain.Entity;

    public interface ILanguagePort
    {
        Task<LanguageEntity> CreateLanguageAsync(LanguageEntity language);
        Task<LanguageEntity> GetLanguageByIdAsync(long id);
        Task<List<LanguageEntity>> GetLanguagesByIdsAsync(List<long> ids);
        Task<LanguageEntity> GetLanguageByCodeAsync(string code);
        Task<LanguageEntity> GetLanguageByNameAsync(string name);
    }
}